### PR TITLE
refactor(typescript): noImplicityAny for index.ts

### PR DESCRIPTION
**Pull request by praveenqlogic**
Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -61,17 +114,28 @@ export interface PushConfig {
   attibutes?: Map<string, string>;
```
Comment (romlogic) [inline src/index.ts]: attributes in line 114 spelled incorrectly.

File: src/index.ts [hunk 2]
```diff
@@ -33,12 +33,12 @@ const PKG = require('../../package.json');
...
 
 import {Snapshot} from './snapshot';
-import {Subscription, SubscriptionMetadata, SubscriptionMetadataRaw} from './subscription';
+import {Subscription, SubscriptionMetadataRaw} from './subscription';
 import {Topic, PublishOptions} from './topic';
 import {CallOptions} from 'google-gax';
 import {Readable} from 'stream';
 import {google} from '../proto/pubsub';
-import {ServiceError} from 'grpc';
+import {ServiceError, ChannelCredentials} from 'grpc';
```
Comment (ajaaym) [inline src/index.ts:41]: I dont think you should use channel credential from grpc instead use credential from google auth

File: src/index.ts [hunk 3]
```diff
@@ -48,6 +48,59 @@ export interface GetSubscriptionMetadataCallback {
...
+  keyFilename?: string;
+  apiEndpoint?: string;
+  email?: string;
+  credentials?: {client_email: string, private_key: string};
+  autoRetry?: boolean;
+  maxRetries?: number;
+  promise?: PromiseConstructor;
+  servicePath?: string;
+  port?: string;
+  sslCreds?: ChannelCredentials;
```
Comment (ajaaym) [inline src/index.ts:62]: why do you need sslcred and credential both? I dont think you need sslcreds.

File: src/index.ts [hunk 4]
```diff
@@ -61,17 +114,28 @@ export interface PushConfig {
   attibutes?: Map<string, string>;
 }
 
+export interface FlowControl {
```
Comment (ajaaym) [inline src/index.ts]: You should export this flow control and batching(bundling) from gax.

File: src/index.ts [hunk 5]
```diff
@@ -61,17 +114,28 @@ export interface PushConfig {
...
+  maxBytes?: number;
+  maxMessages?: number;
+  allowExcessMessages?: boolean;
+}
+
+export interface Batching {
```
Comment (ajaaym) [inline src/index.ts]: You should export this flow control and batching(bundling) from gax.

**Review by ajaaym — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (praveenqlogic) [reply src/index.ts]: Done!, big Thanks :)

File: src/index.ts (hunk 3 above)
Comment (praveenqlogic) [reply src/index.ts:62]: at 297, they mention in comment structure of ClientConfig, there they mention it, so i used, its not used anywhere in application, so removed!

File: src/index.ts (hunk 2 above)
Comment (praveenqlogic) [reply src/index.ts:41]: at line 572, used function return type is ChannelCredentials which is assigning value to sslCreds, thought better to use the same type.

File: src/index.ts (hunk 4 above)
Comment (praveenqlogic) [reply src/index.ts]: tried finding but not able to, i checked with import google-gax module. need little help here.

File: src/index.ts (hunk 5 above)
Comment (praveenqlogic) [reply src/index.ts]: tried finding but not able to, i checked with import google-gax module. need little help here.

File: src/index.ts (hunk 4 above)
Comment (ajaaym) [reply src/index.ts]: keep using this. as pubsub dont use gax flow control and batching. it has its own.

File: src/index.ts (hunk 5 above)
Comment (ajaaym) [reply src/index.ts]: keep using this. as pubsub dont use gax flow control and batching. it has its own.

**Review by ajaaym — CHANGES_REQUESTED** [review]

**Comment by praveenqlogic** [comment]
Completed

**Pull request closed without merging**